src/CONST.ts

```diff
@@ -2890,6 +2890,8 @@ const CONST = {
         LEARN_MORE_LINK: 'https://help.expensify.com/articles/new-expensify/billing-and-plan-types/Referral-Program',
         LINK: 'https://join.my.expensify.com',
     },
+
+    MAX_COUNT_OF_TAGS: 500,
```

Comment:
I think we should set this max for anything that uses this optionList component - categories, tags, etc. Not just tags.

Reply:
So, you mean to set this limit by default? Right now it works like a property we can pass.

Reply:
That's what I was thinking yeah. Though I'm not sure all the places this component is used, so it'd be good to sanity-check that it makes sense.

Reply:
I can find all places and will post it here.

Reply:
So, @puneetlath, the component is [BaseOptionsSelector](https://github.com/Expensify/App/blob/main/src/components/OptionsSelector/BaseOptionsSelector.js) and being used for 8 cases: Category Picker, Tag Picker,  Money Request participant, Split Bill participant, Search, Start Chat. Task Assign, Task Share. 

<details><summary>Screenshoots</summary>

<img width="550" alt="Screenshot 2023-11-22 at 17 08 31" src="https://github.com/Expensify/App/assets/57314631/9295ba27-4589-45b8-9d7e-ff17a0c65dee">


<img width="1728" alt="Category Picker" src="https://github.com/Expensify/App/assets/57314631/143520b9-0c05-4b42-903d-186f1d2e2649">
<img width="1728" alt="Money Request:Split Bill" src="https://github.com/Expensify/App/assets/57314631/c96d61b3-9254-43b0-b806-11e883f8befc">
<img width="1728" alt="Search" src="https://github.com/Expensify/App/assets/57314631/640f469e-adea-4994-baff-69e30bc59f10">
<img width="1728" alt="Start Chat" src="https://github.com/Expensify/App/assets/57314631/7c9d0a68-22ce-427c-baa3-b7bed7124993">
<img width="1728" alt="Tag Picker" src="https://github.com/Expensify/App/assets/57314631/5099e6bb-e27d-4086-86a7-69bb5470c31d">
<img width="1728" alt="Task Assign" src="https://github.com/Expensify/App/assets/57314631/b8164935-2cc8-43f0-b10a-b178e3ac1025">
<img width="1728" alt="Task Share" src="https://github.com/Expensify/App/assets/57314631/362eba31-ca37-456e-8c5b-9f7dc63a67e6">

</details> 

Obviously, I've tried to test the limitation. Actually, I like how it works for all cases. 
Note: for this video I've changed the limitation to 10 items per page. You can also set any value.


https://github.com/Expensify/App/assets/57314631/1d010653-0b5b-43d9-b949-dbe11ce27ad2

Added a commit - b08423a1b6062c5fc1a3fec7f91b48453bbfa216.

cc @rushatgabhane

Reply:
Nice! I agree with you that we should make this limitation the default and use it for all those pages. I don't think there's any reason anyone would need to look at more than 500 results at once for any of them.

Reply:
And it'll be nice that any field that uses the options selector in the future, will automatically have this "show more" button built in. I don't even think we need to make it a prop. We could just make it a feature of that component. If in the future someone wants to make it configurable, then they can add it as a prop.

Reply:
Exactly, agree.

So, updating... Thank you for quick feedback 😉

Reply:
Done - 96c23a4.

Reply:
I love the consistency in behavior across all these screens! 🙌

Reply:
Looks like we are all aligned, so waiting on the review. @rushatgabhane, your turn 😉